### ISSUE TITLE
Feature Request: improvements to logrotate

### ISSUE THREAD

**Issue by maxenglander** [vitessio/vitess#12567]
## Description

There are a few different feature requests, and I thought it might be a bit annoying to give each of these their own GH issue, so putting them all in one for now. Happy to split them out though. I think all of these ideas can be introduced in a way that is backwards compatible for folks who rely on the current behavior of the container.

 - [ ] make gid/uid configurable
 - [ ] add `mysql` binary to the container
 - [ ] make `vtdataroot` is configurable
 - [ ] make log paths, sizes, and rotate params configurable
 - [ ] make standard mysql logs discoverable

## make gid/uid configurable

**problem**

Right now the Dockerfile is creating a `vitess` user and group of 1000 and 2000 respectively. https://github.com/vitessio/vitess/blob/97b151e4b8f96547efb642531c0953de86f065ca/docker/k8s/logrotate/Dockerfile#L30-L31

Meanwhile the `lite` images do not specify a UID/GID, and they end having a UID/GID of 999, at least for me on my Mac M1:

```
$ docker run -it --entrypoint=/bin/bash vitess/lite:latest
vitess@508dc6ec8fbd:/$ cat /etc/passwd | grep vitess
vitess:x:999:999::/home/vitess:/bin/sh
```

The discrepancy in these values makes it difficult to run `docker/lite` and `logrotate` together in a K8s Pod. If we try to set the security context to run the `logrotate` container as user, it will be able to create `logrotate.status`, but won't be able to rotate files owned by user 999. If we set the security context to run as user 999, problem switches the other way.

**recommendation**

Not sure what the intent is with having the `logrotate` UID/GID be 1000/2000. Rather than change this and potentially break things for people who rely on that currently, we can introduce build args `GID` and `UID`. Or we could just break things for people and make these the same as what's in `vitess/lite`.

## add mysql client binary to the container

The current `logrotate.conf` file uses `/usr/bin/mysql` to do some postrotation stuff. But `/usr/bin/mysql` isn't on the container! Not sure how other people are using this currently, but I think it would be helpful if we added the client binary to the container.

https://github.com/vitessio/vitess/blob/97b151e4b8f96547efb642531c0953de86f065ca/docker/k8s/logrotate/logrotate.conf#L1

## make `vtdataroot` is configurable

Not everyone has tablet data located at `/vtdataroot/tabletdata`, which is currently hardcoded in the container. https://github.com/vitessio/vitess/blob/main/docker/k8s/logrotate/logrotate.conf#L1. Would be nice to make this configurable.

## make log paths, sizes, and rotate params configurable

Right now all log files are managed by a wildcard. 

https://github.com/vitessio/vitess/blob/97b151e4b8f96547efb642531c0953de86f065ca/docker/k8s/logrotate/logrotate.conf#L1

It would be nice if `logrotate` allowed changing this wildcard path, and additionally allowing for specifying the paths of commons mysql logs (error, general, slow), and specifying different `size` and `rotate` params for these files.

## make log file paths discoverable

Since `logrotate` is already expected to be able to run `mysql`, it might be nice if it could discover the paths of common mysqld logs (error, general, slow). This is mostly a convenience thing so that users orchestrating Vitess can change the path of say the general log without needing to update the `logrotate` container.

### Use Case(s)

I'm exploring the idea of integrating `logrotate` into our Vitess deployments for cases where we want to turn on the general log, and don't want to use up the entire data volume. The `logrotate` container in its current form is too inflexible at the moment.

**Comment by deepthi** [comment]
@derekperkins are you currently using the logrotate image? What are your thoughts on this proposal?

**Comment by derekperkins** [comment]
Yes, we're using it. I originally wrote it and it was deployed as a part of the helm chart, including specific guarantees about how that worked, including the UID/GID specifics. I'm all for making it more globally useful, I'm most definitely not an expert in the area.
